### PR TITLE
Forward rank profile with templated pluq

### PR DESCRIPTION
**Pull request by ZHG2017**
Having adapted the RankProfile functions that are related to PLUQ which has been restructured wtih ParSeqHelper

### CODE REVIEW

File: fflas-ffpack/ffpack/ffpack.h [hunk 1]
```diff
@@ -1198,6 +1202,13 @@ namespace FFPACK { /* Solutions */
          size_t* P, size_t* Q,
          const FFLAS::FFLAS_DIAG Diag=FFLAS::FflasNonUnit);
 
+    template <class Field, class PSHelper>
+    typename Field::Element&
+    Det( typename Field::Element& det,
```
Comment (ClementPernet) [inline fflas-ffpack/ffpack/ffpack.h]: There should be no Det in this PR. Please remove this change

Comment (ZHG2017) [reply fflas-ffpack/ffpack/ffpack.h]: I saw that some modifications had been added into the master branch then I merged this branch with the recent version of master branch and I see no big difference so I wonder if it is still necessary to revert this Det function.

File: tests/test-rankprofiles.C [hunk 2]
```diff
@@ -167,13 +170,13 @@ bool run_with_field(Givaro::Integer q, uint64_t b, size_t m, size_t n, size_t r,
             // for (size_t i=0; i<r; i++)
             // 	F->assign(RPM[RRP[i]*n+CRP[i]],F->one);
             fassign (*F, n, n, A, lda, B, lda);
-            size_t cs = FFPACK::ColumnRankProfile (*F, n, n, A, lda, CRPLUD, FFPACK::FfpackSlabRecursive);
+            size_t cs = FFPACK::ColumnRankProfile (*F, n, n, A, lda, CRPLUD, FFPACK::FfpackSlabRecursive, seqH);
```
Comment (ClementPernet) [inline tests/test-rankprofiles.C]: Your choice TileRecursive -> parallel, SlabRecursive -> sequential is arbitrary. Please restore all these calls to calls without helper, and add additional calls with parallel helpers after these.

Comment (ClementPernet) [reply tests/test-rankprofiles.C]: Note also that SlabRecursive calls LUdivine which has no parallel version, so you never test the parallel code

File: fflas-ffpack/ffpack/ffpack_rankprofiles.inl [hunk 3]
```diff
@@ -30,8 +30,16 @@ namespace FFPACK{
     template <class Field>
     inline size_t RowRankProfile (const Field& F, const size_t M, const size_t N,
                                   typename Field::Element_ptr A, const size_t lda,
-                                  size_t* &rkprofile,
-                                  const FFPACK_LU_TAG LuTag){
+                                  size_t* &rkprofile, const FFPACK_LU_TAG LuTag){
+        FFLAS::ParSeqHelper::Sequential seqH;
+        size_t R = FFPACK::RowRankProfile (F, M, N, A, lda, rkprofile, LuTag, seqH);
+        return R;
```
Comment (ClementPernet) [inline fflas-ffpack/ffpack/ffpack_rankprofiles.inl]: simpler: return FFPACK::RowRankProfile(....., FFLAS::ParSeqHelper::Sequential());

File: examples/det.C [hunk 4]
```diff
@@ -49,7 +49,7 @@ int main(int argc, char** argv)
     size_t m, n;
     FFLAS::ReadMatrix(file.c_str(), F, m, n, A);
 
-    double d = FFPACK::Det(F, m, n, A, n);
+    double d = FFPACK::Det(F, d, n, A, n);
```
Comment (ClementPernet) [inline examples/det.C]: Good catch. More simply:
{{{
double d;
FFPACK::Det(F,d,n,A,n);
}}}

File: fflas-ffpack/ffpack/ffpack_rankprofiles.inl [hunk 5]
```diff
@@ -30,8 +30,16 @@ namespace FFPACK{
     template <class Field>
```
Comment (ClementPernet) [inline fflas-ffpack/ffpack/ffpack_rankprofiles.inl]: You should apply the same edits to the `ReducedRowRankProfile` and `ReducedColumnRankProfile` routines

File: tests/test-rankprofiles.C (hunk 2 above)
Comment (ZHG2017) [reply tests/test-rankprofiles.C]: I added one sequential helper and one without any helper as before to test if the modification for all possible use of RankProfile but I will revert this if you like

File: fflas-ffpack/ffpack/ffpack_rankprofiles.inl (hunk 5 above)
Comment (ZHG2017) [reply fflas-ffpack/ffpack/ffpack_rankprofiles.inl]: I did not find any of the 2 routines, perhaps you meant either ColumnRankProfile or RowRankProfile, isn't it ?

Comment (ClementPernet) [reply fflas-ffpack/ffpack/ffpack_rankprofiles.inl]: Of course, I got confused between rank profiles and echelon forms!
Forget about this request!

**Comment by ClementPernet** [comment]
Good to go!

**Pull request merged**